### DISCUSSION TITLE
Cannot compile deepmd-kit from source code

### DISCUSSION THREAD

**Discussion by kkkkqq** [deepmodeling/deepmd-kit#1168] [Q&A]
I created a virtual environment, with tensorflow installed via:
conda install tensorflow
I then entered the source directory for deepmd and tried to install deepmd-kit with 
pip install .
Then I got the error:

 -- Disabled cpp interface build, looking for tensorflow_framework
  CMake Error at cmake/Findtensorflow.cmake:153 (message):
    Failed to compile:

     Change Dir: /tmp/pip-req-build-skneh0ib/_skbuild/linux-x86_64-3.9/cmake-build/tf_version/CMakeFiles/CMakeTmp

    Run Build Command(s):/tmp/pip-build-env-7co1bac4/overlay/bin/ninja
    cmTC_dbf78 && [1/2] Building CXX object
    CMakeFiles/cmTC_dbf78.dir/tf_version.cpp.o

    [2/2] Linking CXX executable cmTC_dbf78

    FAILED: cmTC_dbf78

    : && /usr/bin/c++ -std=c++11 -Wno-ignored-attributes
    CMakeFiles/cmTC_dbf78.dir/tf_version.cpp.o -o cmTC_dbf78
    -Wl,-rpath,/mnt/xfs1/home/liutongfei/anaconda3/envs/deepmd/lib/python3.9/site-packages/tensorflow
    /mnt/xfs1/home/liutongfei/anaconda3/envs/deepmd/lib/python3.9/site-packages/tensorflow/libtensorflow_framework.so.2
    && :
.....................................
    /usr/bin/ld: warning: libiomp5.so, needed by
    /mnt/xfs1/home/liutongfei/anaconda3/envs/deepmd/lib/python3.9/site-packages/tensorflow/libtensorflow_framework.so.2,
    not found (try using -rpath or -rpath-link)
  ERROR: Failed building wheel for deepmd-kit
Failed to build deepmd-kit
ERROR: Could not build wheels for deepmd-kit which use PEP 517 and cannot be installed directly
with a very long error message

I have tried upgrading pip, it didn't work.

Then I tried installing the package using conda. I tried installing version 2.0.0 with gpu using the command below:
 conda create -n deepmdgpu deepmd-kit=2.0.0=*gpu libdeepmd=2.0.0=*gpu lammps-dp cudatoolkit=10.1 -c https://conda.deepmodeling.org
It worked this time, running alright with the water demo. However, I would need to compile from source code. I'm therefore wondering what caused this issue, and how to avoid it?

**Comment by amcadmus** [comment]
plz use pip to install tensorflow

**Comment by njzjz** [comment]
libiomp5.so is intel openmp library and can be provided by https://anaconda.org/anaconda/intel-openmp
